build.py

```diff
@@ -203,6 +205,28 @@ def set_config(filename, config, new_value):
             fp.write(line)
 
 
+def is_kernel_header_installed():
+    try:
+        cmd = 'dpkg-query -W -f \'${Status}\' linux-headers-%s' \
```

Comment:
Won't this change make build.py ubuntu dependent, which up to this point it has not been? 

Can this be driven from the KERNELDIR shell variable as is done in core/kmod/Makefile. If KERNELDIR is set or if 
/lib/modules/$(uname -r)/build exist, then build the modules.

Note, I do like the existing method whereby a local build environment can edit dpdk-17.05-common_linux as they see fit for their environment. I use IGB_UIO but I do not use KNI and would prefer not to build it.

Reply:
You are right. No dependency would be definitely better. I saw so many helper scripts using apt-get, but I have noticed that they are not in the critical path.